bokeh/model.py

```diff
@@ -741,7 +727,21 @@ def __exit__(self, type, value, traceback):
 
     def __enter__(self):
         for model in self._to_remove_after:
-            self._doc.add_root(model)
+            try:
+                self._doc.add_root(model)
+            except RuntimeError as e:
+                try:
+                    for r in takewhile(lambda x: x != model, self._to_remove_after):
+                        r.document.remove_root(r)
+                except RuntimeError:
+                    pass
```

Comment:
I think the `try/except` might be safest inside the loop. I recognize that the list is ordered but if someone later changes it to a set then it might not be, it's just more defensive to make sure all the roots to remove are all always visited.